src/inference/src/dev/core_impl.cpp

```diff
@@ -1473,21 +1486,46 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
                 update_config[ov::loaded_from_cache.name()] = true;
 
                 if (util::contains(plugin.get_property(ov::supported_properties), ov::weights_path)) {
-                    std::string weights_path = cacheContent.modelPath;
-                    auto pos = weights_path.rfind('.');
-                    if (pos != weights_path.npos && weights_path.substr(pos) == ".xml") {
-                        weights_path = weights_path.substr(0, pos);
-                        weights_path += ".bin";
+                    util::Path weights_path;
+
+                    if (auto&& path_hint = update_config.find(ov::weights_path.name());
+                        path_hint != update_config.end()) {
+                        weights_path = path_hint->second.as<std::string>();
+                    } else if (weights_path = extract_weight_path(header.get_runtime_info()); weights_path.empty()) {
+                        weights_path = cacheContent.modelPath;
+                        weights_path.replace_extension(".bin");
                     }
+
                     if (ov::util::file_exists(weights_path)) {
-                        update_config[ov::weights_path.name()] = weights_path;
+                        update_config[ov::weights_path.name()] = weights_path.string();
                     }
                 }
                 if (model_buffer) {
-                    update_config[ov::internal::cached_model_buffer.name()] = model_buffer;
+                    auto compiled_blob =
+                        Tensor(element::from<char>(), Shape{model_buffer->size()}, model_buffer->get_ptr());
```

Comment:
`element::from<char>()` is a `ov::element::boolean`. and it confuses a little.
Maybe use u8 for this purpose?

Reply:
I think this code can be replaced if Tensor from file will be supported. The char is use to allow directly cast to char for shared buffer. But it depends on #29175 implementation then this code can be synced.

Reply:
But anyway by Tensor(element::from<char>(), ...) we get tensor of bools. It looks unnatural. There is another such tensor in template plugin

Reply:
Ok, will update it.